### PR TITLE
HADOOP-16637. Fix findbugs warnings in hadoop-cos.

### PR DESCRIPTION
**Pull request by cxorm**
## NOTICE

Please create an issue in ASF JIRA before opening a pull request,
and you need to set the title of the pull request which starts with
the corresponding JIRA issue number. (e.g. HADOOP-XXXXX. Fix a typo in YYY.)
For more details, please see https://cwiki.apache.org/confluence/display/HADOOP/How+To+Contribute

### CODE REVIEW

File: hadoop-cloud-storage-project/hadoop-cos/src/main/java/org/apache/hadoop/fs/cosn/CosNativeFileSystemStore.java [hunk 1]
```diff
@@ -249,7 +250,15 @@ public void storeEmptyFile(String key) throws IOException {
 
   public PartETag uploadPart(File file, String key, String uploadId,
       int partNum) throws IOException {
-    InputStream inputStream = new FileInputStream(file);
+    InputStream inputStream = null;
```
Comment (steveloughran) [inline hadoop-cloud-storage-project/hadoop-cos/src/main/java/org/apache/hadoop/fs/cosn/CosNativeFileSystemStore.java]: looks broken, as L262 will always get a closed stream

**Comment by steveloughran** [comment]
Most of the changes look good; that input stream cleanup looks like it will break the code as a closed stream will always get passed along to the next method. The findbugs file probably just needs changing.

Which endpoint did you run the tests against?

**Comment by cxorm** [comment]
> Most of the changes look good; that input stream cleanup looks like it will break the code as a closed stream will always get passed along to the next method. The findbugs file probably just needs changing.
> 
> Which endpoint did you run the tests against?

We just ran the findbugs for the file.

**Comment by steveloughran** [comment]
ok. You need to run the full test suite against a live service and declare which location/endpoint you tested with

that's the policy for all object store patches I'm afraid -Yetus can't be given the credentials and yet we need to stop regressions creeping in. Sorry

**Comment by cxorm** [comment]
Thanks @steveloughran for the comment.
I'm going to run the related tests.

File: hadoop-cloud-storage-project/hadoop-cos/src/main/java/org/apache/hadoop/fs/cosn/BufferPool.java [hunk 2]
```diff
@@ -86,10 +85,6 @@ private File createDir(String dirPath) throws IOException {
       } else {
         LOG.debug("buffer dir: {} already exists.", dirPath);
       }
-    } else {
-      throw new IOException("creating buffer dir: " + dir.getAbsolutePath()
-          + "unsuccessfully.");
-    }
```
Comment (aajisaka) [inline hadoop-cloud-storage-project/hadoop-cos/src/main/java/org/apache/hadoop/fs/cosn/BufferPool.java]: Would you fix indent?

Comment (cxorm) [reply hadoop-cloud-storage-project/hadoop-cos/src/main/java/org/apache/hadoop/fs/cosn/BufferPool.java]: Thanks @aajisaka for the review.
Yes I would fix it, and ran the test on this issue.

**Comment by aajisaka** [comment]
Hi @cxorm, how is this issue going?

**Comment by cxorm** [comment]
Thanks @aajisaka for the review.
And we test the patch with our cluster for at least 30 days without any error.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |  24m 36s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  19m 26s |  trunk passed  |
| +1 :green_heart: |  compile  |   0m 24s |  trunk passed  |
| +1 :green_heart: |  checkstyle  |   0m 21s |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 26s |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  14m 53s |  branch has no errors when building and testing our client artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 25s |  trunk passed  |
| +0 :ok: |  spotbugs  |   0m 38s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| -1 :x: |  findbugs  |   0m 36s |  hadoop-cloud-storage-project/hadoop-cos in trunk has 5 extant findbugs warnings.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 21s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 16s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 16s |  the patch passed  |
| -0 :warning: |  checkstyle  |   0m 13s |  hadoop-cloud-storage-project/hadoop-cos: The patch generated 3 new + 0 unchanged - 0 fixed = 3 total (was 0)  |
| +1 :green_heart: |  mvnsite  |   0m 18s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedclient  |  13m 40s |  patch has no errors when building and testing our client artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 21s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   0m 40s |  hadoop-cloud-storage-project/hadoop-cos generated 0 new + 1 unchanged - 4 fixed = 1 total (was 5)  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 19s |  hadoop-cos in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 31s |  The patch does not generate ASF License warnings.  |
|  |   |  79m 16s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.8 Server=19.03.8 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1640/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1640 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 35751b260277 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 3d5ade1 |
| Default Java | 1.8.0_242 |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1640/2/artifact/out/branch-findbugs-hadoop-cloud-storage-project_hadoop-cos-warnings.html |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-1640/2/artifact/out/diff-checkstyle-hadoop-cloud-storage-project_hadoop-cos.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1640/2/testReport/ |
| Max. process+thread count | 414 (vs. ulimit of 5500) |
| modules | C: hadoop-cloud-storage-project/hadoop-cos U: hadoop-cloud-storage-project/hadoop-cos |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1640/2/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by aajisaka** [comment]
Hi @cxorm there is still 1 findbugs warning, would you check this?
https://builds.apache.org/job/hadoop-multibranch/job/PR-1640/2/artifact/out/patch-findbugs-hadoop-cloud-storage-project_hadoop-cos-warnings.html
In addition, would you fix the following checkstyle issues?
https://builds.apache.org/job/hadoop-multibranch/job/PR-1640/2/artifact/out/diff-checkstyle-hadoop-cloud-storage-project_hadoop-cos.txt

Sorry for the late response.

**Pull request closed without merging**